src/ssm.c

```diff
@@ -113,21 +113,36 @@ static void Check_Error(BMS_INPUT_T *input, BMS_STATE_T *state, BMS_OUTPUT_T *ou
     // }
     //[TODO] change to case statement
     if (state->curr_mode != BMS_SSM_MODE_INIT) {
-        int16_t max_cell_temp_thres_C = state->pack_config->max_cell_temp_dC;
```

Comment:
FSAE believes (and I assumed you did as well) that a fault should only cause the contactors to open if said fault persists for the pre-specified length of time (or appears the pre-specified number of times in the case of counting faults), which are defined in error_handler.c. 

The old behavior of the code in ssm.c, unless I'm misunderstanding, was as follows. Consider the case where cell X goes under the minimum voltage at time t=K milliseconds. As soon as that voltage drop was detected by the system, `Error_Assert(CELL_UNDER_VOLTAGE);` would be executed, which would store time K. 

Imagine then that at time K+10, cell X goes back over the minimum voltage and never again goes under. `Error_Pass(CELL_UNDER_VOLTAGE)` would never be executed, and so the time K would still be saved for the `CELL_UNDER_VOLTAGE` error and the .error flag would still be true, even though the `Error_Assert` command was no longer being executed. 

Finally, at time K+1000, `Error_Handle` in main.c would determine that the error timeout of 1000ms had completed and would throw a CELL_UNDER_VOLTAGE fault, causing a FORCED_HANG. We believe that this is incorrect behavior, as the error should have instead been cleared when it disappeared before the timeout expired.

However, the current PR is also not optimal, as a fault that persists for 999ms repeatedly and has a single ms of the fault not being present will never trigger an error. If we detect that our sensors are unreliable and will occasionally mis-read, frequently enough to never trigger these faults, then we should modify the logic in error_handler.c to add a third class of error: an "aggregate timeout" of sorts. This aggregate timeout would fire if a fault is observed for a total time of Y ms in addition to a continuous time of X, where Y >> X. However this is a more complex refactor that I believe we should only do if we realize that faulty sensors are a frequent occurrence.

Let me know if this helps.